clients/unshaded/src/main/java/tachyon/client/block/TachyonBlockStore.java

```diff
@@ -168,4 +169,23 @@ public long getUsedBytes() throws IOException {
       mContext.releaseMasterClient(blockMasterClient);
     }
   }
+
+  public void promote(long blockId) throws IOException {
+    BlockMasterClient blockMasterClient = mContext.acquireMasterClient();
+    try {
+      BlockInfo info = blockMasterClient.getBlockInfo(blockId);
+      if (info.getLocations().isEmpty()) {
+        throw new IOException("Block " + blockId + " is not available in Tachyon.");
```

Comment:
Actually is throwing an exception really good here? When we are calling `promote(myBlockId)`, do we always assume this block is in Tachyon space?